library/entropy.c

```diff
@@ -333,33 +336,45 @@ int mbedtls_entropy_func( void *data, unsigned char *output, size_t len )
     memset( buf, 0, MBEDTLS_ENTROPY_BLOCK_SIZE );
 
 #if defined(MBEDTLS_ENTROPY_SHA512_ACCUMULATOR)
-    mbedtls_sha512_finish( &ctx->accumulator, buf );
+    if( ( ret = mbedtls_sha512_finish_ext( &ctx->accumulator, buf ) ) != 0 )
+        goto exit;
 
     /*
      * Reset accumulator and counters and recycle existing entropy
      */
     memset( &ctx->accumulator, 0, sizeof( mbedtls_sha512_context ) );
-    mbedtls_sha512_starts( &ctx->accumulator, 0 );
```

Comment:
@andresag01 You already mentioned this issue lately: Doing a `memset` on the `sha512_context accumulator` breaks the abstraction and should be replaced by a call ty `mbedtls_sha512_init`. Did you already do that in another pending PR, or shall we to the changes here?